src/simulator.zig

```diff
@@ -120,6 +120,20 @@ pub fn main() !void {
     const node_count = replica_count + standby_count;
     const client_count = 1 + random.uintLessThan(u8, constants.clients_max);
 
+    const batch_size_limit_min = comptime batch_size_limit_min: {
+        var event_size_max: u32 = @sizeOf(vsr.RegisterRequest);
+        for (std.enums.values(StateMachine.Operation)) |operation| {
+            const event_size = @sizeOf(StateMachine.Event(operation));
+            event_size_max = @max(event_size_max, event_size);
+        }
+        break :batch_size_limit_min event_size_max;
+    };
+    const batch_size_limit: u32 = if (random.boolean())
+        constants.message_body_size_max
+    else
+        batch_size_limit_min +
+            random.uintAtMost(u32, constants.message_body_size_max - batch_size_limit_min);
```

Comment:
Ideally, we'd check that a cluster with _different_ batch sizes behaves as expected. 

We actually can do it the same way we do with upgrades? Ratcheting batch_size up over time?

Reply:
(see https://github.com/tigerbeetle/tigerbeetle/pull/1981#issuecomment-2161295191)